pkg/securelaunch/helpers.go

```diff
@@ -12,26 +12,45 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
+	"sync"
 
-	"github.com/u-root/u-root/pkg/boot/diskboot"
+	"github.com/u-root/u-root/pkg/mount"
 	"github.com/u-root/u-root/pkg/storage"
 )
 
-/* used to store all block devices returned from a call to storage.GetBlockStats */
-var storageBlkDevices []storage.BlockDev
+type persistDataItem struct {
+	desc        string // Description
+	data        []byte
+	location    string // of form sda:/path/to/file
+	defaultFile string // if location turns out to be dir only
+}
+
+var persistData []persistDataItem
+
+type mountCacheData struct {
+	flags     uintptr
+	mountPath string
+}
+
+type mountCacheType struct {
+	m  map[string]mountCacheData
+	mu sync.RWMutex
+}
+
+// mountCache is used by sluinit to reduce number of mount/unmount operations
+var mountCache = mountCacheType{m: make(map[string]mountCacheData)}
```

Comment:
some of the stuff in this file (mount cache, storage blk devices) is pretty nice -- we should perhaps generalize it into pkg/storage or pkg/mount (I can do that if you want).

Reply:
Thanks. Sure. I can definitely add it to pkg/storage or pkg/mount although I am afraid I might not be able to test this as well on other platforms since I am only using these in securelaunch package and that too on X7 (or X8 in future) platform...
How about this...we let these changes in this PR go as-is under securelaunch pkg. That way our QA team can start testing it and it wont disturb their delivery time line...
And then we can  see how to move this to pkg/storage or pkg/mount in a separate PR which has minimum effect (in terms of bugs) on other users...? thoughts ?